### PR TITLE
Expansion of AliAnalysisTaskEmcalJetEnergyFlow.cxx to handle MC data on both levels

### PR DESCRIPTION
**Pull request by ChrisPliatskas**
This commit contains many changes on AliAnalysisTaskEmcalJetEnergyFlow analysis task which are the following: a) Additional histograms were included, e.g. the Constituent Z vs jet pt as well as the Response-matrix (on Dpt and pt direction).b) The task has been expanded to handle the analysis of MC data, particularly analyse jets at the generator and detector level within the same run. c)Lastly there have been some binning adjustments for the EnergyFlow histograms as well as some improvements on the code-readability side.

### CODE REVIEW

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h [hunk 1]
```diff
@@ -41,7 +41,8 @@ class AliAnalysisTaskEmcalJetEnergyFlow: public AliAnalysisTaskEmcalJet {
 		const char *ntracks		= "usedefault",
 		const char *nclusters		= "usedefault",
 		const char *ncells		= "usedefault",
-		const char *suffix		= "");
+		const char *suffix		= "",
+                Bool_t isMCprod                 = kTRUE               );
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h:45]: Your suffix should be last if you want to support subwagons

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h [hunk 2]
```diff
@@ -66,6 +67,7 @@ class AliAnalysisTaskEmcalJetEnergyFlow: public AliAnalysisTaskEmcalJet {
 	void                    DoClusterLoop()                         ; ///<May remove later	
 	void                    DoCellLoop()                            ; ///<May remove later
 
+        Bool_t                  isMCprod                                ;/// Flag for MC productions
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h:70]: If you're streaming this (it's ambiguous from your comment, but based on the header, I think you want to stream it), then you need to bump the classdef

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h (hunk 1 above)
Comment (ChrisPliatskas) [reply PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h]: Thank you for the valuable feedback! I think that I incorporated the changes you suggested ( and unless I messed up my with my git) the latest PR should include these changes.

Comment (raymondEhlers) [reply PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetEnergyFlow.h]: Thanks, it looks good. Approving now

**Comment by raymondEhlers** [comment]
+1

**Pull request merged**